### PR TITLE
Added NumericInputBox edit support

### LINKED ISSUE #16: WebEdit: Add Number support
Support for `ENumberInputField` using the numeric input control

### PR DESCRIPTION
**Pull request by vzhovtyuk**
Fixed #16. Added Type conversion Double to Integer and BigDecimal.

### CODE REVIEW

File: de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/builders/EYesNoRadioBuilder.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 import org.apache.poi.util.StringUtil;
 
 /**
- * Defines the InputBox builder class.
+ * Defines the YesNoRadioGroup builder class.
  * 
  * @author Aron Cotrau
```
Comment (lippisch) [inline de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/builders/EYesNoRadioBuilder.java]: I think if you copy a class, you should also update the author to your name ;-)

File: de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/NumericInputboxMapper.java [hunk 2]
```diff
@@ -16,27 +16,28 @@
...
+import de.xwic.appkit.core.dao.ValidationResult.Severity;
 import de.xwic.appkit.core.model.queries.IPropertyQuery;
 import de.xwic.appkit.webbase.editors.ValidationException;
-import de.xwic.appkit.webbase.editors.builders.EYesNoRadioBuilder;
 
 /**
- * Mapper for the InputBox control.
+ * Mapper for the NumericInputBox control.
  *
  * @author Aron Cotrau
```
Comment (lippisch) [inline de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/NumericInputboxMapper.java]: Again - Change the author to yourself - or delete the tag if you want to remain anonymous

File: de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/PropertyMapper.java [hunk 3]
```diff
@@ -275,6 +276,15 @@ protected void writeValue(IEntity entity, Property[] property, Object value) thr
 				} else if (targetType.equals(Double.class)) {
 					value = iValue.doubleValue();
 				}
+			} else if (value instanceof Double) { // Integer to..
```
Comment (lippisch) [inline de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/PropertyMapper.java]: really minor.. but another small copy & paste glitch... its.. // Double to...

File: de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/YesNoRadioGroupMapper.java [hunk 4]
```diff
@@ -53,6 +53,9 @@ public void loadContent(IEntity entity, RadioGroup text, Property[] property) th
         if(value instanceof Boolean) {
             final Boolean booleanValue = (Boolean) value;
             text.setSelectedKey(booleanValue ? EYesNoRadioBuilder.KEY_YES : EYesNoRadioBuilder.KEY_NO);
+        } else if(value instanceof String) {
+            final String strValue = (String) value;
+            text.setSelectedKey(strValue);
```
Comment (lippisch) [inline de.xwic.appkit.webbase/src/de/xwic/appkit/webbase/editors/mappers/YesNoRadioGroupMapper.java]: I am not sure how "safe" this is. You may end up with setting a key that doesnt exist. How about something like this:

```java
text.setSelectedKey("1".equals(value) || "true".equals("value") ? EYesNoRadioBuilder.KEY_YES : ...KEY_NO);
```

**Comment by lippisch** [comment]
it's a good implementation so far, just minor things. Please review and let me know when adjusted.

**Comment by vzhovtyuk** [comment]
Comments addressed. Please review.

**Pull request merged**